dvc/output/base.py

```diff
@@ -293,7 +295,12 @@ def commit(self):
             self.cache.save(self.path_info, self.cache.tree, self.hash_info)
 
     def dumpd(self):
-        ret = copy(self.hash_info.to_dict())
+        ret = {}
+
+        if not self.IS_DEPENDENCY and self.desc:
+            ret[self.PARAM_DESC] = self.desc
```

Comment:
Ugly because Output/DependencyBase needs a refactor (will try to send one after this PR). But the jist is that we're setting desc first, as it is more intuitive to have before the hash.